src/State/DelayScheduler.lua

```diff
@@ -0,0 +1,73 @@
+--!strict
+
+--[[
+  Manages batch updating of delay objects.
+]]
+
+local RunService = game:GetService("RunService")
+
+local Package = script.Parent.Parent
+local Types = require(Package.Types)
+local updateAll = require(Package.Dependencies.updateAll)
+
+type Array<T> = {[number]: T}
+type Delay = Types.Delay<any>
+type DelayData = {
+  delay: Delay,
+  start: number,
+  duration: number,
+  goal: any
+}
+
+local DelayScheduler = {}
+
+local WEAK_KEYS_METATABLE = {__mode = "k"}
+
+-- all delays to be updated
+local allDelays: Array<DelayData> = {}
```

Comment:
Order is not important, so this seems like it should be a Set.

Reply:
Fixed in 2e0c77a